lib/worker/OperationQueue.js

```diff
@@ -81,19 +83,18 @@ module.exports = class OperationQueue {
       });
     }
 
-    // if the queue is not initialized or the head of the queue is at the end
-    // of the queue, then no next chunk of operations cached yet, so create one
-    // that will be cached when `getNextChunk` is called and its operations are
-    // retrieved
-    if(!(Array.isArray(this.queue) && (this.head + 1) < this.queue.length)) {
+    // if the queue is not initialized with the head of the queue before the end
+    // of the queue, then no next chunk of operations is cached yet, so create
+    // one that will be cached when `getNextChunk` is called and its operations
+    // are retrieved
+    const end = (this.queue || []).length - 1;
+    if(!(Array.isArray(this.queue) && this.head < end)) {
```

Comment:
Can we simplify to an existence check on `this.queue`? It's never anything but `null`/`undefined` and an `Array`, right?

```suggestion
    if(!(this.queue && this.head < end)) {
```